### PR TITLE
PL-156 Knowledge graph components refactor & PL-155 Pdf reader components refactor

### PR DESCRIPTION
**Pull request by NinaGnjidic**
PL-156 Knowledge graph components refactor
Done:
- refactored KnowledgeGraphComponent (replaced then with async-await)
- refactored SidePageComponent so that this component handles displaying/hiding KnowledgeGraphComponent and SummaryKeyComponent
- added hide, show and isOpened methods to KnowledgeGraphComponent and SummaryKeyComponent which SidePageComponent calls when handling displaying/hiding of those components (maybe better option is to have a reference to html containers of those components in SidePageComponent and remove those methods from those components - **need opinion on this**)
- removed unused imports and formatted PaperInfoComponent
- moved SidePageComponent from PdfReaderComponent to ReaderView
- created new event onReadNewPdf which is published when reader reads new document and this event can then be used in further refactoring (instead of calling reset method of components in PdfReaderComponent)
- made hideLoader method in SidePageLoaderComponent public so SidePageComponent can call it without publishing an event (maybe it's better to use event and keep this method private since that event is already being used in KnowledgeGraphComponent  - **need opinion on this**)
-html of SummaryKeyComponent is now contained in html of SidePageComponent 
- new cursor while knowledge graph is loading
- fixed cursor in SummaryKeyComponent

TODO:
- close knowledge graph/summary when reference page is opened and vice versa
- write tests
- refactor further?

### CODE REVIEW

**Review by tonesija — COMMENTED** [review]
I would like to see what has exactly been done in the PR message.
What are those pdf.worker files in dist?

**Comment by gabrielekarra** [comment]
if i open the kg, then close it i can't open it again until i open and close the summary

and idk what is this dot 
image

if i click on a reference and two page layout is displayed, if i click the kg button the reference close, but no kd displayed

**Comment by NinaGnjidic** [comment]
> if i open the kg, then close it i can't open it again until i open and close the summary
> 
> and idk what is this dot image
> 
> if i click on a reference and two page layout is displayed, if i click the kg button the reference close, but no kd displayed

You're right, first thing was a bug which was happening because I did not implement closing side page correctly.
That blue dot was there because zindex of depth selector was too high.

In my last two commits both should be fixed, so you can check it out again.

**Comment by NinaGnjidic** [comment]
> I would like to see what has exactly been done in the PR message. What are those pdf.worker files in dist?

Okok I should've done that, I'll update PR's message.

**Comment by NinaGnjidic** [comment]
> Screenshot 2023-01-04 alle 09 43 34
> 
> If the knowledge graph cannot be generated the page is compressed on the top and the error isn't shown. The rest seems to be all fine

This does not happen on my machine, could be that some style is not applied on IOS but I don't know how can I test this out.
image

File: src/components/ReferenceViewComponent.js [hunk 1]
```diff
@@ -4,161 +4,143 @@ import {
...
-    main: document.querySelector("#main"),
-    pdfDoc: null,
-    pdfPageNumber: null,
-    viewport: null,
-    graphMakerBtn: document.querySelector("#graph-maker"),
-    sideNav: document.querySelector("#side-page"),
-    closeBtnReference: document.createElement("button"),
+    canvas: document.querySelector("#reference-canvas"),
+    closeBtn: document.querySelector("#reference-close-btn"),
+    container: document.querySelector("#reference-container"),
```
Comment (Giamburrasca) [inline src/components/ReferenceViewComponent.js:23]: I refactor this components too on the PL 127, let's pay attention when merging

**Review by Giamburrasca — COMMENTED** [review]
Really great and huge work!!

**Comment by Giamburrasca** [comment]
> > Screenshot 2023-01-04 alle 09 43 34
> > If the knowledge graph cannot be generated the page is compressed on the top and the error isn't shown. The rest seems to be all fine
> 
> This does not happen on my machine, could be that some style is not applied on IOS but I don't know how can I test this out. image

I have a Mac too and it works even displaying the error page.

**Review by tonesija — APPROVED** [review]
Where do those pdf.worker.sdasd come from?

+1, beautiful

**Review by matteovisotto — APPROVED** [review]
Really a beautiful work. +1

**Pull request merged**